test.js

```diff
@@ -290,6 +290,35 @@ function runTests(options) {
         });
       }));
     });
+    it('should watch removed and re-added directories', function(done) {
+      var unlinkSpy = sinon.spy(function unlinkSpy(){});
+      var addSpy = sinon.spy(function addSpy(){});
+      var parentPath = getFixturePath('subdir');
+      var subPath = getFixturePath('subdir/dir');
+      watcher
+        .on('unlinkDir', unlinkSpy)
+        .on('addDir', addSpy)
+        .on('ready', d(function() {
+          fs.mkdirSync(parentPath);
+          d(function() {
+            fs.rmdirSync(parentPath);
+            d(function() {
+              fs.mkdirSync(parentPath);
+              d(function() {
```

Comment:
Those values are fairly arbitrary - based on what seemed to work consistently for the other tests. You're free to use `setTimeout` directly if it works out more cleanly for this particular situation.